python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -606,16 +716,54 @@ def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationRe
                 # rematerializations from causing a chain of materializations to be kicked off
                 ignored_parent_keys={context.asset_key},
             )
-            updated_parents = {parent.asset_key for parent in updated_parent_asset_partitions}
-            will_update_parents = will_update_parents_by_asset_partition[asset_partition]
+            for parent in updated_parent_asset_partitions:
+                asset_partitions_by_updated_parents[parent].add(asset_partition)
 
-            if updated_parents or will_update_parents:
-                asset_partitions_by_evaluation_data[
-                    ParentUpdatedRuleEvaluationData(
-                        updated_asset_keys=frozenset(updated_parents),
-                        will_update_asset_keys=frozenset(will_update_parents),
-                    )
-                ].add(asset_partition)
+            for parent in parent_asset_partitions:
+                if context.will_update_asset_partition(parent):
+                    asset_partitions_by_will_update_parents[parent].add(asset_partition)
+
+        filtered_updated_and_will_update_parents = self.updated_parent_filter.passes(
```

Comment:
is it significantly more expensive to do this filtering check inside the above loop? (i'd believe you if you said yes, but if not then a bunch of the complexity below here could be cut out as there's a bunch of shuffling and stuff going on)

Reply:
ah I see that you said that it does in the description

Reply:
yeah was quite slow :(